packages/core/src/transport/httpRequest.ts

```diff
@@ -13,26 +15,55 @@ import { addTelemetryError } from '../domain/telemetry'
 export type HttpRequest = ReturnType<typeof createHttpRequest>
 
 export function createHttpRequest(endpointBuilder: EndpointBuilder, bytesLimit: number) {
-  return {
-    send: (data: string | FormData, bytesCount: number) => {
-      const url = endpointBuilder.build()
-      const canUseBeacon = !!navigator.sendBeacon && bytesCount < bytesLimit
-      if (canUseBeacon) {
-        try {
-          const isQueued = navigator.sendBeacon(url, data)
-
-          if (isQueued) {
-            return
-          }
-        } catch (e) {
-          reportBeaconError(e)
+  function sendBeacon(data: string | FormData, bytesCount: number) {
+    const url = endpointBuilder.build()
+    const canUseBeacon = !!navigator.sendBeacon && bytesCount < bytesLimit
+    if (canUseBeacon) {
+      try {
+        const isQueued = navigator.sendBeacon(url, data)
+
+        if (isQueued) {
+          return
         }
+      } catch (e) {
+        reportBeaconError(e)
       }
+    }
+
+    sendXHR(url, data)
+  }
+
+  function sendFetch(data: string | FormData, bytesCount: number) {
+    const url = endpointBuilder.build()
+    const canUseKeepAlive = window.Request && 'keepalive' in new Request('') && bytesCount < bytesLimit
+    if (canUseKeepAlive) {
+      fetch(url, { method: 'POST', body: data, keepalive: true }).catch(
+        monitor(() => {
+          // failed to queue the request
+          sendXHR(url, data)
+        })
+      )
+    } else {
+      sendXHR(url, data)
+    }
+  }
+
+  function sendXHR(url: string, data: string | FormData) {
+    const request = new XMLHttpRequest()
+    request.open('POST', url, true)
+    request.send(data)
+  }
 
-      const request = new XMLHttpRequest()
-      request.open('POST', url, true)
-      request.send(data)
+  return {
+    send: (data: string | FormData, bytesCount: number) => {
+      // TODO check tests without flag
+      if (!isExperimentalFeatureEnabled('fetch-keepalive')) {
+        sendBeacon(data, bytesCount)
+      } else {
+        sendFetch(data, bytesCount)
+      }
     },
+    sendBeacon,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏This method is a bit confusing because it might not even use `sendBeacon` at all.  Maybe rename this method (and related methods in `Batch`) to something more explicit like `sendUrgently` (since it is used when leaving the page)

Reply:
yep, not super fan of the naming as well.

> since it is used when leaving the page

`sendOnExit`?

Reply:
Sounds good to me!